src/server/player_sao.cpp

```diff
@@ -457,20 +457,25 @@ u16 PlayerSAO::punch(v3f dir,
 
 void PlayerSAO::setHP(s32 hp, const PlayerHPChangeReason &reason)
 {
-	s32 oldhp = m_hp;
+	if (hp == (s32)m_hp)
+		return; // Nothing to do
 
-	hp = rangelim(hp, 0, m_prop.hp_max);
+	if (m_hp <= 0 && hp < (s32)m_hp)
```

Comment:
Equivalent to `if (m_hp <= 0 && hp < 0)`, or `if (!m_hp && hp < 0)`

Reply:
`hp < (s32)m_hp` should clarify that it's only about handling damage, which would not be directly obvious with functionally equivalent expressions.